test/test_enzyme.jl

```diff
@@ -267,3 +267,83 @@ end
     rel_error = abs(dc²_dκ[1][3] - dc²_dκ_fd) / abs(dc²_dκ_fd)
     @test  rel_error < tol
 end
+
+
+@testset "Enzyme on advection and diffusion WITH flux boundary condition" begin
+    Nx = Ny = 64
+    Nz = 8
+
+    Lx = Ly = L = 2π
+    Lz = 1
+
+    x = y = (-L/2, L/2)
+    z = (-Lz/2, Lz/2)
+    topology = (Periodic, Periodic, Bounded)
+
+    grid = RectilinearGrid(size=(Nx, Ny, Nz); x, y, z, topology)
+    diffusion = VerticalScalarDiffusivity(κ=0.1)
+
+    u = XFaceField(grid)
+    v = YFaceField(grid)
+
+    U = 1
+    u₀(x, y, z) = - U * cos(x + L/8) * sin(y) * (z + L/2)
+    v₀(x, y, z) = + U * sin(x + L/8) * cos(y) * (z + L/2)
+
+    set!(u, u₀)
+    set!(v, v₀)
+    fill_halo_regions!(u)
+    fill_halo_regions!(v)
+
+    @inline function tracer_flux(x, y, t, c, p)
```

Comment:
suggest using the discrete form because most real cases will use that.

Reply:
I switched this test to discrete form and it now passes.